sys/dev/mfi/mfi_ioctl.h

```diff
@@ -87,7 +87,15 @@ struct mfi_ioc_packet {
 	} mfi_frame;
 
 	struct iovec mfi_sgl[MAX_IOCTL_SGE];
-} __packed __aligned(sizeof(void *__capability));
+}
+#if !(defined(_KERNEL) || defined(__CHERI_PURE_CAPABILITY__))
```

Comment:
This condition is not quite right, you want something closer to what `__kerncap` uses, that is either a CHERI-aware kernel or a CPC userland, so something like:

```
#if !(((defined(_KERNEL) && __has_features(capability)) || defined(__CHERI_PURE_CAPABILITY__))
```

One wonders if we want to define a helper __CHERI_USER_ABI in some central <sys/*.h> header (perhaps next to __kerncap) that we can use in various places rather than duplicating this complicated expression (I think we are on the order of close to 10 copies of this #if now)